documentation/src/main/markdown/packagemgrs/opam.md

```diff
@@ -0,0 +1,37 @@
+# Opam Support
+
+## Related properties
+
+[Detector properties](../properties/detectors/opam.md)
+
+[detect_product_short] has two detectors for Opam:
+
+* OPAM CLI Detector
+* OPAM Lock Detector
+
+## OPAM CLI Detector
+
+* This detectable executes opam commands to discover dependencies of opam projects.
+
+* This OPAM detector will be executed on your project if [detect_product_short] finds `.opam` file in your top level directory. It requires `opam`
+exe to be present on your $PATH. You can also override the location for `opam` exe by the OPAM path property.
+
+The OPAM Build Detector will work in the following way on your project:
+
+1. [detect_product_short] OPAM Build Detector will run `opam --version` to get version of opam on your machine.
+2. If the version of opam is greater than or equal to 2.2.0, then [detect_product_short] would run `opam tree . --with-test --with-doc --with-dev --recursive`
+to get list of resolved packages installed in the current switch for the project.
+<note type="note">User must have all the pre-requisites for the project already set up on your machine (e.g the opam switch where your packages for project are installed)
+before running [detect_product_short] on your installed.</note>
```

Comment:
Minor:
`<note type="note">You must have all prerequisites for the project set up on your machine (e.g., the opam switch where your packages for the project are installed), before running [detect_product_short].</note>`